### PR TITLE
add auth endpoints and decode_jwt

### PR DESCRIPTION
**Pull request by Teasotea**

### CODE REVIEW

File: backend/routes/user.py [hunk 1]
```diff
@@ -4,30 +4,63 @@
...
+    user_query["meals"] = ''
+
+    request.app.database.collection.insert_one(user_query)
+    return {'token': encoded_jwt, "username": user_query["username"], "message": "User registered successfully"}
+
+
+@router.post("/logout")
+@router.options("/logout")
+async def logout_user(request: Request):
+    authorization = request.headers.get("Authorization")
```
Comment (ruslankotliar) [inline backend/routes/user.py:55]: U have to remove "Bearer" from header.
"Authorization" header will look like this: 'Bearer someauthorizationtoken'

Read [this](https://security.stackexchange.com/questions/108662/why-is-bearer-required-before-the-token-in-authorization-header-in-a-http-re).

File: backend/routes/user.py [hunk 2]
```diff
@@ -4,30 +4,63 @@
...
+    if existing_user:
+        return {"message": "User already exists"}
+    
+    hashed_password = hashlib.sha256(user_query["password"].encode()).hexdigest()
+    encoded_jwt = jwt_auth.get_encoded_jwt(user_query)
+
+    user_query["password"] = hashed_password
+    user_query["token"] = encoded_jwt
+    user_query["ingredients"] = ''
+    user_query["meals"] = ''
```
Comment (ruslankotliar) [inline backend/routes/user.py:46]: It is better not to store jwt in DB. The idea is to validate user jwt using cryptography, but not compare it to DB data.

File: backend/routes/user.py [hunk 3]
```diff
@@ -4,30 +4,63 @@
...
+
+    request.app.database.collection.insert_one(user_query)
+    return {'token': encoded_jwt, "username": user_query["username"], "message": "User registered successfully"}
```
Comment (ruslankotliar) [inline backend/routes/user.py:49]: I do not need message field here.

**Review by ruslankotliar — DISMISSED** [review]

File: backend/routes/user.py (hunk 2 above)
Comment (YukihiroSM) [reply backend/routes/user.py:46]: that was a suggestion just for now. Later we will improve that.

File: backend/routes/user.py (hunk 3 above)
Comment (YukihiroSM) [reply backend/routes/user.py:49]: @ruslankotliar We should also return user_id from the database, right?

**Review by YukihiroSM — APPROVED** [review]

**Pull request merged**